server/src/resolvers/wrapResolversWithAuth.ts

```diff
@@ -0,0 +1,156 @@
+
+/**
+ * 
+ * Wrap a specific resolver with the auth wrapper resolver
+ * This has the actual auth prototype implementation
+ * 
+ * The wrapResolversWithAuth function below calls this function
+ * for each resolver
+ * 
+ * @param resolverAuthInfo the config for a specific resolver
+ * @param resolveFn the original resolver function
+ * 
+ */
+function wrapResolverWithAuth(resolverAuthInfo: any, resolveFn: any): any {
```

Comment:
This is the wrapping resolvers approach. Going with the wrapping approach means we should be able to follow one single approach that covers most (if not all of the auth cases), whereas going with the CrudService approach will probably require some level of wrapping in addition.

There is a benefit of this approach - it's fairly easy to understand and follow.

The big drawback I see is that wrapping resolvers is a little weird and might not play nicely if the user wants to do some of their own wrapping. Looking for feedback on this.